drivers/mm/mm_drv_intel_adsp_imr_memoryblock.c

```diff
@@ -0,0 +1,83 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+/*
+ * Copyright (c) 2022 Intel Corporation.
+ *
+ * Author: Marcin Szkudlinski <marcin.szkudlinski@linux.intel.com>
+ *
+ */
+
+#define DT_DRV_COMPAT intel_adsp_imr
+
+#include <zephyr/device.h>
+#include <zephyr/kernel.h>
+#include <zephyr/sys/mem_blocks.h>
+#include <zephyr/linker/linker-defs.h>
+#include <zephyr/drivers/mm/mm_drv_intel_adsp_imr_memoryblock.h>
+
+#include <soc.h>
+
+/* get IMR parameters from the device tree */
+#define IMR_REGION DT_DRV_INST(0)
+#define IMR_BASE_ADDR LINKER_DT_RESERVED_MEM_GET_PTR(DT_DRV_INST(0))
+#define IMR_MEMORY_SIZE LINKER_DT_RESERVED_MEM_GET_SIZE(DT_DRV_INST(0))
+#define IMR_PAGE_SIZE DT_PROP(DT_DRV_INST(0), block_size)
+#define IMR_NUM_OF_PAGES (IMR_MEMORY_SIZE / IMR_PAGE_SIZE)
+
+/* declare an IMR memory block */
+SYS_MEM_BLOCKS_DEFINE_WITH_EXT_BUF(
+		IMR_REGION,
+		IMR_PAGE_SIZE,
+		IMR_NUM_OF_PAGES,
+		(uint8_t *) IMR_BASE_ADDR);
```

Comment:
Does this driver get built at all? Compiler should have complained about duplicate identifier `IMR_REGION` because the mem_block is named `IMR_REGION` and you also have a `#define IMR_REGION` above.